### PR TITLE
Settings provider

### PR DESCRIPTION
**Pull request by rozenbergidan**

### CODE REVIEW

File: lib/views/components/settings/settings_content.dart [hunk 1]
```diff
@@ -1,10 +1,156 @@
...
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return Center(child: Text("im settings screen"),);
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Column(children: [
+        Row(
```
[the remark this answers is not in the document]
Comment (rozenbergidan) [reply lib/views/components/settings/settings_content.dart:77]: done

File: lib/views/components/settings/settings_content.dart [hunk 2]
```diff
@@ -1,10 +1,156 @@
-import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:numberpicker/numberpicker.dart';
+import 'package:safety_eye_app/providers/settings_provider.dart';
 
-class SettingsPage extends StatelessWidget {
-  const SettingsPage({super.key});
+import '../../../services/preferences_services.dart';
+
+const double fontSize = 12;
```
[the remark this answers is not in the document]
Comment (rozenbergidan) [reply lib/views/components/settings/settings_content.dart:7]: done

File: lib/views/screens/home_screen.dart [hunk 3]
```diff
@@ -25,8 +28,13 @@ class _HomeScreenState extends State<HomeScreen> {
 
   @override
   Widget build(BuildContext context) {
+    settingsProvider = Provider.of<SettingsProvider>(context);
+    _pages = [RecordingPage(), JourneysPage(), SettingsPage(settingsProvider)];
```
[the remark this answers is not in the document]
Comment (rozenbergidan) [reply lib/views/screens/home_screen.dart:32]: moved to be init in the main app and drilled down

File: lib/models/Settings.dart [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  PreferencesKeys.publicKey: 'default_public_key',
+  PreferencesKeys.initializeKeys: false,
+  PreferencesKeys.chunkDuration: 60,
+  PreferencesKeys.autoUpload: false,
+  PreferencesKeys.gracePeriodInterval: 20,
+  PreferencesKeys.videoResolution: 'high',
+};
+
+class Settings {
+  late String privateKey;
```
[the remark this answers is not in the document]
Comment (rozenbergidan) [reply lib/models/Settings.dart:14]: done

File: lib/views/components/settings/settings_content.dart [hunk 5]
```diff
@@ -1,10 +1,156 @@
...
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Column(children: [
+        Row(
+          children: [
+            Text('Chunk duration (seconds)'),
+            Spacer(),
+            NumberPicker(
+              value: chunkDuration,
+              minValue: 40,
```
[the remark this answers is not in the document]
Comment (rozenbergidan) [reply lib/views/components/settings/settings_content.dart:83]: I think it not going to be used, and I wouldn't wont to put so many named variables - thus wont fix

**Review by EdanLevy — APPROVED** [review]
LGTM

**Pull request merged**